### PR TITLE
See API Rate limit

### PR DESCRIPTION
**Pull request by montogeek**
Third take.

### CODE REVIEW

**Comment by montogeek** [comment]
@jeremenichelli @EugeneHlushko @byzyk  I did a rather quick refactoring to fetch scripts, so I can debug GH API Rate limit easily, as you can see in the build logs, API Token is working.

**Comment by byzyk** [comment]
Nice, let's see what causing the limit to exceed. Very likely it could be Travis issue.

**Comment by byzyk** [comment]
@montogeek just to clarify, GH token is set as Travis env variable, right?

**Comment by montogeek** [comment]
@byzyk Yes

File: src/scripts/fetch_package_names.js [hunk 1]
```diff
@@ -12,6 +19,8 @@ if (require.main === module) {
 function main() {
   const organization = process.argv[2];
   const suffix = process.argv[3];
+  const file = process.argv[4];
+  const output = process.argv[5];
 
   if (!organization) {
```
Comment (EugeneHlushko) [inline src/scripts/fetch_package_names.js]: this is not for now but i would like to see an iteration vs 4 ifs

File: src/scripts/fetch_package_names.js [hunk 2]
```diff
@@ -65,3 +105,98 @@ function fetchPackageNames(options, cb) {
...
+    function(pkg, cb) {
+      const branch = 'master';
+      const url = ['https://raw.githubusercontent.com', pkg.full_name, branch, file].join('/');
+
+      request(url, function(err, response, body) {
+        if (err) {
+          return cb(err);
+        }
+
+        if (body && file === 'README.md') {
```
Comment (EugeneHlushko) [inline src/scripts/fetch_package_names.js]: I think we had this in other PR but is it safe to assume that all the newly added webpack-contrib package readme files will be named exactly `README.md` and not `readme.md` or `README.MD` etc?

**Comment by EugeneHlushko** [comment]
Guess it should work, sorry i couldnt test now

File: src/scripts/fetch_package_names.js [hunk 3]
```diff
@@ -43,6 +78,11 @@ function fetchPackageNames(options, cb) {
       type: 'token',
       token: process.env.GITHUB_TOKEN
     });
+
+    github.misc.getRateLimit({}, (err, res) => {
```
Comment (jeremenichelli) [inline src/scripts/fetch_package_names.js]: What's the goal of this besides probably debugging on CI?

File: src/scripts/fetch_package_names.js [hunk 4]
```diff
@@ -65,3 +105,98 @@ function fetchPackageNames(options, cb) {
...
+          }
+        );
+      });
+    },
+    finalCb
+  );
+}
+
+// TODO: push this type of to a script of its own to keep this generic
+function yamlHeadmatter(fields) {
```
Comment (jeremenichelli) [inline src/scripts/fetch_package_names.js]: Don't we have a util that already generates this based on a set of information? I think I did separate this logic, would love if you could adapt it so we don't go back to tech debt around this;

File: src/scripts/fetch_package_names.js (hunk 3 above)
Comment (montogeek) [reply src/scripts/fetch_package_names.js]: That is the goal

File: src/scripts/fetch_package_names.js (hunk 4 above)
Comment (montogeek) [reply src/scripts/fetch_package_names.js]: will try

Comment (montogeek) [reply src/scripts/fetch_package_names.js]: @jeremenichelli `rebuild` branch probably, it is not in `master`

File: src/scripts/fetch_package_names.js (hunk 2 above)
Comment (montogeek) [reply src/scripts/fetch_package_names.js]: No

**Comment by montogeek** [comment]
@EugeneHlushko @jeremenichelli Any comments about this?

File: src/scripts/fetch.sh [hunk 5]
```diff
@@ -8,14 +8,14 @@ mkdir -p ./generated/plugins
 cp -rf ./src/content/plugins/ ./generated/plugins
 
 # Fetch webpack-contrib (and various other) loader repositories
-node ./src/scripts/fetch_package_names.js "webpack-contrib" "-loader" | node ./src/scripts/fetch_package_files.js "README.md" "./generated/loaders"
-node ./src/scripts/fetch_package_names.js "babel" "babel-loader" | node ./src/scripts/fetch_package_files.js "README.md" "./generated/loaders"
-node ./src/scripts/fetch_package_names.js "postcss" "postcss-loader" | node ./src/scripts/fetch_package_files.js "README.md" "./generated/loaders"
-node ./src/scripts/fetch_package_names.js "peerigon" "extract-loader" | node ./src/scripts/fetch_package_files.js "README.md" "./generated/loaders"
+node ./src/scripts/fetch_package_names.js "webpack-contrib" "-loader" "README.md" "./generated/loaders"
```
Comment (EugeneHlushko) [inline src/scripts/fetch.sh]: i guess the last thing i would like changed is the `fetch_package_names.js` which becamse `fetch_packages` after merging with `fetch_package_files.js` functionality, makes sense to me to rename it accordingly

File: src/scripts/fetch_package_names.js (hunk 4 above)
Comment (jeremenichelli) [reply src/scripts/fetch_package_names.js]: Oh sorry I thought this was on the rebuild one. Fine then :)

File: src/scripts/fetch.sh (hunk 5 above)
Comment (montogeek) [reply src/scripts/fetch.sh]: Done

**Pull request merged**